README.md

```diff
@@ -34,14 +34,236 @@ VSCode allows for development within a docker container. This project has an alr
 
 -[Developing inside a Container](https://code.visualstudio.com/docs/remote/containers) -[Docker](https://www.docker.com/)
 
+
+### Prerequisites
+
+Make sure the following dependencies are installed on your system:
+- Poetry (1.5.1)
+- Visual Studio Code
+- Docker
+- MangoDB Compass
+- 1Password
+
 ## Getting Started
 
-### To Begin Development
+Follow these steps to set up the development environment after installing the prerequisites.
+1. Setting up GitHub SSH Auth and Signing Keys and configure SSH authentication and add your SSh public key to your GitHub account.
+    - Open 1 pass, and in the right top corner click on new items.
+    - Then click on SSH keys, and then add private key.
+    - Then click on generate private key.Select an SSH Key > Generate New Key.
+    - Select SSH key type: Ed25529, then click generate
+    - Once, you got the key copy the public key and go to your github account.
+    - Go to settings, click on SSH and GPG keys and click on New SSH Keys.
+    - Once you pasted the key, click on key type and click on Authentication.
+    - Repeat the steps, and but instead of an Authentication key type, now click signing key.
+    - Open your SSH client configuration file (~/.ssh/config or C:\Users\YourUsername\.ssh\config).
+    - Then create your ~/.ssh folder or config file by using this command :
+       $ export SSH_AUTH_SOCK=~/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock
+    - Then add the IdentityAgent snippet to your ~/.ssh/config file:
+        - Host *
+            IdentityAgent "~/Library/Group Containers/2BUA8C4S2C.com.1password/t/agent.sock"
+2. Cloning the Repository:
+    - clone the repository using SSH:
```

Comment:
You should probably clarify that you are copying the repo's remote URL in GitHub itself. Code > Clone > SSH